velox/expression/FunctionSignature.h

```diff
@@ -109,6 +109,8 @@ class SignatureVariable {
 
 class FunctionSignature {
  public:
+  /// Construct a function signature. Need to call the validate method to
```

Comment:
Does all the constructor should follow validate method? If so, I would suggest to add a new function `construct` to call constructor and validate

Reply:
virtual method cannot be used in constructor.

Reply:
We have `FunctionSignatureBuilder` and `AggregateFunctionSignatureBuilder`, we can construct `FunctionSignature` solely through their build methods. This way, we only need to ensure that the `validate` method is called within the build method. I will modify the code in `AggregateCompanionSignatures.cpp` that directly calls the constructor to use the build method for construction instead.